src/i18n/injectIntlWithShim.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { injectIntl, intlShape } from 'react-intl';
+
+
+const injectIntlWithShim = (WrappedComponent) => {
+  class ShimmedIntlComponent extends React.Component {
+    static propTypes = {
+      intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+      super(props);
+      this.shimmedIntl = Object.create(this.props.intl, {
+        formatMessage: {
+          value: (definition) => {
+            if (definition === undefined) {
```

Comment:
Can we add a check here to also make sure that it's an object with an id property?